webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala

```diff
@@ -765,50 +724,50 @@ object Authenticator extends InstrumentationSupport {
-                             case None      =>
-                               // should not happen, as the token is already validated
-                               throw AuthenticationException(
-                                 "No IRI found inside token. Please report this as a possible bug."
-                               )
-                           }
-                           // log.debug("getUserADMThroughCredentialsV2 - used session token")
-                           getUserByIdentifier(
-                             identifier = UserIdentifierADM(maybeIri = Some(userIri))
-                           )
-                         case None =>
-                           // log.debug("getUserADMThroughCredentialsV2 - no credentials supplied")
-                           throw BadCredentialsException(BAD_CRED_NONE_SUPPLIED)
-                       }
+      user <- credentials match {
+                case Some(passCreds: KnoraPasswordCredentialsV2) =>
+                  // log.debug("getUserADMThroughCredentialsV2 - used identifier: {}", passCreds.identifier)
+                  getUserByIdentifier(
+                    identifier = passCreds.identifier
+                  )
+                case Some(KnoraJWTTokenCredentialsV2(jwtToken)) =>
+                  val userIri: IRI = JWTHelper.extractUserIriFromToken(
+                    jwtToken,
+                    settings.jwtSecretKey,
+                    settings.externalKnoraApiHostPort
+                  ) match {
+                    case Some(iri) => iri
+                    case None      =>
+                      // should not happen, as the token is already validated
+                      throw AuthenticationException(
+                        "No IRI found inside token. Please report this as a possible bug."
+                      )
+                  }
+                  // log.debug("getUserADMThroughCredentialsV2 - used token")
+                  getUserByIdentifier(
+                    identifier = UserIdentifierADM(maybeIri = Some(userIri))
+                  )
+                case Some(KnoraSessionCredentialsV2(sessionToken)) =>
+                  val userIri: IRI = JWTHelper.extractUserIriFromToken(
+                    sessionToken,
+                    settings.jwtSecretKey,
+                    settings.externalKnoraApiHostPort
+                  ) match {
+                    case Some(iri) => iri
+                    case None      =>
+                      // should not happen, as the token is already validated
+                      throw AuthenticationException(
+                        "No IRI found inside token. Please report this as a possible bug."
+                      )
+                  }
+                  // log.debug("getUserADMThroughCredentialsV2 - used session token")
+                  getUserByIdentifier(
+                    identifier = UserIdentifierADM(maybeIri = Some(userIri))
+                  )
```

Comment:
When is this case used?

Reply:
when a cookie is used for authentication. the cookie is a header that holds the JWT token. The other cases are KnoraJWTtokenCredentialsV2 which is also a JWT token but sent in the Authentication header. We always use JWT.